### PR TITLE
feat: Support feature flag evaluation server side

### PR DESCRIPTION
**Pull request by mutdmour**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: packages/cli/src/UserManagement/UserManagementHelper.ts [hunk 1]
```diff
@@ -162,6 +163,31 @@ export function sanitizeUser(user: User, withoutKeys?: string[]): PublicUser {
...
+		}, 1500);
+	});
+
+	const fetchPromise = new Promise<CurrentUser>(async (resolve) => {
+		user.featureFlags = await postHog.getFeatureFlags(user);
+
+		resolve(user);
+	});
+
+	return Promise.race([fetchPromise, timeoutPromise]);
```
Comment (OlegIvaniv) [inline packages/cli/src/UserManagement/UserManagementHelper.ts:188]: Ha, love this workaround!

**Review by OlegIvaniv — CHANGES_REQUESTED** [review]
Looking good! 👏  Left just a few minor comments

File: packages/editor-ui/src/stores/posthog.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	const isVariantEnabled = (experiment: string, variant: string) => {
+		return getVariant(experiment) === variant;
+	};
+
+	const identify = () => {
+		const instanceId = rootStore.instanceId;
+		const user = usersStore.currentUser;
+		const traits: Record<string, string | number> = { instance_id: instanceId };
+
+		// todo check why Date is used there
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/stores/posthog.ts:39]: fixed.. removed Date type from this. thanks

File: packages/editor-ui/src/stores/posthog.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		} else if (user && typeof user.createdAt === 'string') {
+			traits.created_at_timestamp = new Date(user.createdAt).getTime();
+		}
+
+		// For PostHog, main ID _cannot_ be `undefined` as done for RudderStack.
+		let id = user ? `${instanceId}#${user.id}` : instanceId;
+		window.posthog?.identify?.(id, traits);
+	};
+
+	const init = (bootstrapped?: FeatureFlags) => {
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/stores/posthog.ts:51]: will rename to something clearer

File: packages/editor-ui/src/stores/posthog.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+				EXPERIMENTS_TO_TRACK.forEach(trackExperiment);
+			}, 0);
+		}
+	);
+
+	watch(
+		() => usersStore.currentUserId,
+		(userId, prevId) => {
+			if (!userId && prevId) {
+				onLogout();
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/stores/posthog.ts:125]: alright will make the change

[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/stores/posthog.ts:125]: went back and forth this. might be clearer if all posthog related code is together.. but it's simpler to understand without the watch. will keep for now

[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/stores/posthog.ts]: nvm moved it to logout

**Review by MiloradFilipovic — APPROVED** [review]
Looks to me all revew comments are addressed.

**Comment by janober** [comment]
Got released with `n8n@0.217.1`

**Pull request merged**